engine/src/ast/field_expr.rs

```diff
@@ -255,21 +258,26 @@ impl<'s> Expr<'s> for FieldExpr<'s> {
         }
 
         match self.op {
-            FieldOp::IsTrue => lhs.compile_with(move |x| cast_value!(x, Bool)),
-            FieldOp::Ordering { op, rhs } => {
-                lhs.compile_with(move |x| op.matches_opt(x.strict_partial_cmp(&rhs)))
-            }
+            FieldOp::IsTrue => lhs.compile_with(false, move |x| cast_value!(x, Bool)),
+            FieldOp::Ordering { op, rhs } => match op {
+                OrderingOp::NotEqual => {
+                    lhs.compile_with(true, move |x| op.matches_opt(x.strict_partial_cmp(&rhs)))
```

Comment:
I am still puzzled by this. Wireshark meaning of every comparison is something like `if any value for the field where the comparison is true, then evaluates to true else evaluates to false`. In this scenario it makes sense to evaluates to false for nil.